File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/UninitializedWalletException.java
package com.github.dedis.popstellar.utility.error.keys;

import com.github.dedis.popstellar.R;

public class UninitializedWalletException extends KeyException {

  public UninitializedWalletException() {
    super("The wallet is not initialized");
  }

  @Override
  public int getUserMessage() {
    return R.string.uninitzilized_wallet_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/UninitializedWalletException.kt
package com.github.dedis.popstellar.utility.error.keys

import com.github.dedis.popstellar.R

class UninitializedWalletException : KeyException("The wallet is not initialized") {
    override val userMessage: Int
        get() = R.string.uninitialized_wallet_exception
    override val userMessageArguments: Array<Any?>
        get() = arrayOfNulls(0)
}